File: Prog_Lang/hxb1638_lab01/hxb_lab01.py
#Hunter Befort
#1001181638
#Python 3.9.6
#macOS Ventura 13.5.1

import os

def dirSize(dir):
    sum = 0
    for entry in os.scandir(dir):           #parses through the directorty
        if entry.is_file():                 #if next item in directory is a file...
            sum += os.path.getsize(entry)       #adds the current file size to the total directory size
        elif entry.is_dir():                #if the next item in the directory is a subdirectory...
            sum += dirSize(entry.path)          #recurse through the subdirectory and add the total sub size to the total directory size
    return sum

print("Size of directory: " + str(dirSize(".")) + " bytes")

File: Prog_Lang/hxb1638_lab01/hxb_lab01.java
//Hunter Befort
//1001181638
//OpenJDK 17.0.5
//macOS Ventura 13.5.1

import java.io.File;

public class hxb_lab01
{
    public static long dirSize(File dir)
    {
        long sum = 0;
        for(File item : dir.listFiles())        //parses through all files in the directory
        {
            if(item.isFile())                   //if the current item is a file...
            {
                sum += item.length();               //add the size of it to the directory size sum
            }
            else if(item.isDirectory())         //if the current item is a subdirectory...
            {
                sum += dirSize(item);               //recurse through the subdirectory, add its total size to the directory size sum
            }
        }
        return sum;
    }

    public static void main(String[] args)
    {
        File dir = new File(".");
        System.out.println("Size of directory: " + dirSize(dir) + " bytes");
    }
}